container/theme_test.go

```diff
@@ -0,0 +1,37 @@
+package container
+
+import (
+	"image"
+	"testing"
+
+	"fyne.io/fyne/v2/test"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestThemeOverride_Icons(t *testing.T) {
+	b := widget.NewButtonWithIcon("", theme.HomeIcon(), func() {})
+	o := NewThemeOverride(b, test.Theme())
+	w := test.NewWindow(o)
+	plain := w.Canvas().Capture().(*image.NRGBA)
+
+	o.Theme = test.NewTheme()
+	o.Refresh()
+	changed := w.Canvas().Capture().(*image.NRGBA)
+
+	assert.NotEqual(t, plain.Pix, changed.Pix)
```

Comment:
Tests only containing `NotEqual` are always a little strange. I think an image comparison or at least a markup comparison would have been better here (and in the text test below).

Reply:
> at least a markup comparison would have been better here (and in the text test below).

Unfortunately a markup comparison won't show the differences because the widget state is identical, it is the theme context applied that changes. So no markup difference - only render pixel colour.

Reply:
Then, we should add this.

Reply:
Added image tests as requested